torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -221,13 +221,19 @@ void pushPackingPastRnn(Block *b) {
     if(rnn->owningBlock() != n->owningBlock())
       continue;
 
-    // The rnn is followed by a reshape (if bidirectional), and then
-    // by a squeeze (always).
+    // The rnn is followed by a transpose and a reshape (if
+    // bidirectional), and then by a squeeze (always).
     Node * next = rnn->outputs()[0]->uses()[0].user;
-    Node * squeeze = next->kind() == onnx::Reshape
-      ? next->outputs()[0]->uses()[0].user
-      : next
-      ;
+    Node * squeeze = nullptr;
+    if (next->kind() == onnx::Transpose) {
+      Node * reshape = next->outputs()[0]->uses()[0].user;
+      if (reshape->kind() != onnx::Reshape) {
+        continue;
+      }
+      squeeze = reshape->outputs()[0]->uses()[0].user;
+    } else {
+      squeeze = next;
+    }
```

Comment:
In which case this would probably be
```
// The rnn is followed by a transpose and a reshape (if
// bidirectional) or a squeeze (otherwise).
Node * next = rnn->outputs()[0]->uses()[0].user;
if (next->kind() == onnx::Transpose) {
  next = next->outputs()[0]->uses()[0].user;
  if (next->kind() != onnx::Reshape) {
    continue;
  }
} else {
  if (next->kind() != onnx::Squeeze) {
    continue;
  }
}
```
with `squeeze` replaced by `next` in the rest of `pushPackingPastRNN`.